includes/wp-admin/class-health-check.php

```diff
@@ -334,4 +301,82 @@ public static function debug_information( $info ) {
 
 		return $info;
 	}
+
+	/**
+	 * Threaded Comments tests.
+	 *
+	 * @return array The test result.
+	 */
+	public static function test_threaded_comments() {
+		$result = array(
+			'label'       => \__( 'Threaded (nested) comments enabled', 'activitypub' ),
+			'status'      => 'good',
+			'badge'       => array(
+				'label' => \__( 'ActivityPub', 'activitypub' ),
+				'color' => 'green',
+			),
+			'description' => \sprintf(
+				'<p>%s</p>',
+				\__( 'Your threaded (nested) comments are enabled and working correctly.', 'activitypub' )
+			),
+			'actions'     => '',
+			'test'        => 'test_threaded_comments',
+		);
+
+		if ( '1' !== get_option( 'thread_comments', '0' ) ) {
+			$result['status']         = 'recommended';
+			$result['label']          = \__( 'Threaded (nested) comments are not enabled', 'activitypub' );
+			$result['badge']['color'] = 'orange';
+			$result['description']    = \sprintf(
+				'<p>%s</p><p>%s</p><p>%s</p><p>%s</p>',
+				\__( 'We recommend enabling threaded (nested) comments to improve the user experience and engagement on your site and on the fediverse.', 'activitypub' ),
+				\__( 'When threaded comments are disabled, all comments appear in a flat list without any visual connection between replies. This makes it harder for users to follow conversations, especially when someone replies to a specific comment. Enabling threaded comments helps create a clear visual hierarchy of the discussion.', 'activitypub' ),
```

Comment:
Aw man I had the perfect feedback sentence here before GitHub lost it after it asked me to refresh because of a trunk merge :( 

Anyway, I think we can lose this sentence and the part about improving user experience in the one above. The setting is necessary for the consumption of Replies to work correctly in the Fediverse, not because we're concerned about user experience on the blog. I think the description should reflect that